cmd/appengine/device.go

```diff
@@ -1634,32 +1635,42 @@ func parseSendDataPayload(payload string, mappingType interfaces.AstarteMappingT
 	case interfaces.BinaryBlobArray, interfaces.BooleanArray, interfaces.DateTimeArray, interfaces.DoubleArray,
 		interfaces.IntegerArray, interfaces.LongIntegerArray, interfaces.StringArray:
 
-		//wait it's all string?
-		payload = strings.Replace(payload, "[", "", -1)
-		payload = strings.Replace(payload, "]", "", -1)
-		payload_parsed := strings.Split(payload, ",")
-		//always has been
+		//check if payload is an empty array, bypass conversions and just return an empty array
+		// if it is not, proceed as usual
+		if payload == "[]" {
+			ret = make([]interface{}, 0)
 
-		jsonOut := make([]interface{}, len(payload_parsed))
-		for i, v := range payload_parsed {
-			jsonOut[i] = v
-		}
+		} else {
 
-		retArray := []interface{}{}
-		// Do a smarter conversion here.
-		for _, v := range jsonOut {
-			switch val := v.(type) {
-			case string:
-				p, err := parseSendDataPayload(strings.TrimSpace(val), interfaces.AstarteMappingType(strings.TrimSuffix(string(mappingType), "array")))
-				if err != nil {
-					return nil, err
+			//wait it's all string?
+			payload = strings.Replace(payload, "[", "", -1)
+			payload = strings.Replace(payload, "]", "", -1)
+			payload_parsed := strings.Split(payload, ",")
+			//always has been
+
+			jsonOut := make([]interface{}, len(payload_parsed))
+			for i, v := range payload_parsed {
+				jsonOut[i] = v
+			}
+
+			retArray := []interface{}{}
+
+			// Do a smarter conversion here.
+			for _, v := range jsonOut {
+				switch val := v.(type) {
+				case string:
+					p, err := parseSendDataPayload(strings.TrimSpace(val), interfaces.AstarteMappingType(strings.TrimSuffix(string(mappingType), "array")))
+					if err != nil {
+						return nil, err
+					}
+					retArray = append(retArray, p)
+				default:
+					retArray = append(retArray, val)
```

Comment:
Do we ever reach this point? I'm not sure, and in case we hit this line we'd append a `v.(type)` in retArray (that looks strange).
Is it reasonable assuming that within the `parseSendDataPayload`, when dealing with arrays, we always deal with strings? A double check is needed here. In case that assumption is true, we could remove the `switch` construct.
@Annopaolo opinions here?

Reply:
> when dealing with arrays, we always deal with strings? 
looks like that's the case, command line always passes a string and that's why this problem arise

Reply:
A [type assertion](https://go.dev/tour/methods/15) could be used then, as @matt-mazzucato suggested